python/cudf/cudf/core/indexed_frame.py

```diff
@@ -150,24 +149,14 @@
 )
 
 
-def _get_host_unique(array):
+def _get_unique_drop_labels(array):
+    """Return labels to be dropped for IndexFrame.drop."""
     if isinstance(array, (cudf.Series, cudf.Index, ColumnBase)):
-        return array.unique.to_pandas()
-    elif isinstance(array, (str, numbers.Number)):
-        return [array]
+        yield from as_column(array).unique().values_host
```

Comment:
> These are columns and not rows, right?

Yeah these are just column _labels_ to drop that happen to be GPU backed. 

I agree it might be worth doing this on the CPU instead. I'd assume `len(columns to drop) << len(columns)` and we convert to host anyways to iterate over these labels to drop, so we might as well do the `unique` step there too